server/base/src/main/java/org/apache/accumulo/server/metadata/ConditionalTabletMutatorImpl.java

```diff
@@ -251,6 +241,12 @@ public Ample.ConditionalTabletMutator requireSame(TabletMetadata tabletMetadata,
   public void submit(Ample.RejectionHandler rejectionCheck) {
     Preconditions.checkState(updatesEnabled, "Cannot make updates after calling mutate.");
     Preconditions.checkState(sawOperationRequirement, "No operation requirements were seen");
+    if (checkPrevEndRow && extent.prevEndRow() != null) {
```

Comment:
Need to check the prevEndRow when its null.  The `encodePrevEndRow()` method handles null and is what is used to serialize the prev end row for writes.

```suggestion
    if (checkPrevEndRow) {
```